src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseExceptionThrowHelpers.cs

```diff
@@ -0,0 +1,526 @@
+                if (stringIsNullOrEmpty is null || stringLength is null || stringEmpty is null || getType is null)
+                {
+                    return;
+                }
+
+                // Get the ThrowXx helpers on those types.  Some of these may not exist.
+                // If we don't have any such helpers available, there's nothing to do.
+                ISymbol? aneThrowIfNull = ane.GetMembers("ThrowIfNull").FirstOrDefault();
+                ISymbol? aeThrowIfNullOrEmpty = ae.GetMembers("ThrowIfNullOrEmpty").FirstOrDefault();
+                ISymbol? odeThrowIf = ode.GetMembers("ThrowIf").FirstOrDefault();
+                ISymbol? aooreThrowIfZero = aoore.GetMembers("ThrowIfZero").FirstOrDefault();
+                ISymbol? aooreThrowIfNegative = aoore.GetMembers("ThrowIfNegative").FirstOrDefault();
+                ISymbol? aooreThrowIfNegativeOrZero = aoore.GetMembers("ThrowIfNegativeOrZero").FirstOrDefault();
+                ISymbol? aooreThrowIfGreaterThan = aoore.GetMembers("ThrowIfGreaterThan").FirstOrDefault();
+                ISymbol? aooreThrowIfGreaterThanOrEqual = aoore.GetMembers("aooreThrowIfGreaterThanOrEqual").FirstOrDefault();
+                ISymbol? aooreThrowIfLessThan = aoore.GetMembers("ThrowIfLessThan").FirstOrDefault();
+                ISymbol? aooreThrowIfLessThanOrEqual = aoore.GetMembers("ThrowIfLessThanOrEqual").FirstOrDefault();
+                if (aneThrowIfNull is null && aeThrowIfNullOrEmpty is null && odeThrowIf is null &&
+                    aooreThrowIfZero is null && aooreThrowIfNegative is null && aooreThrowIfNegativeOrZero is null &&
+                    aooreThrowIfGreaterThan is null && aooreThrowIfGreaterThanOrEqual is null &&
+                    aooreThrowIfLessThan is null && aooreThrowIfLessThanOrEqual is null)
+                {
+                    return;
+                }
+
+                // If we have any of the ArgumentOutOfRangeException.Throw methods, we're likely to have all of them.
+                bool hasAnyAooreThrow =
+                    aooreThrowIfZero is not null || aooreThrowIfNegative is not null || aooreThrowIfNegativeOrZero is not null ||
+                    aooreThrowIfGreaterThan is not null || aooreThrowIfGreaterThanOrEqual is not null ||
+                    aooreThrowIfLessThan is not null || aooreThrowIfLessThanOrEqual is not null;
+
+                // Look for throw operations.
+                context.RegisterOperationAction(context =>
+                {
+                    var throwOperation = (IThrowOperation)context.Operation;
+
+                    // Try to get the exception object creation operation. As a heuristic, avoid recommending replacing
+                    // any exceptions where a meaningful message may have been provided.  This is an attempt to reduce
+                    // false positives, at the expense of potentially more false negatives in cases where a non-valuable
```

Comment:
do analyzers have something like an optional "informational" (or "pedantic") mode where it could flag all of these rejects for those who want to inspect them manually?

Reply:
In general, no; the general mechanism for that is using different diagnostic IDs that default to different levels of severity.  You can implement various forms of configuration to analyzers, e.g. some design-related analyzers let you specify the visibility of methods you want them to apply to.

Reply:
We should be able to add an editorconfig option for switching to the more aggressive mode. Likely want to wait for feedback/customer requests before doing so.